examples/models/export_hf_model.py

```diff
@@ -0,0 +1,178 @@
+    ("THUDM/chatglm3-6b",),
+]
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-hfm",
+        "--hf_model_repo",
+        required=False,
+        default=None,
+        help="a valid huggingface model repo name",
+    )
+    parser.add_argument("-o", "--output-dir", default=".", help="output directory")
+    parser.add_argument(
+        "-etx",
+        "--to_et_w_xnn",
+        required=False,
+        action="store_true",
+        help="export ExecuTorch with XNNPACK",
+    )
+    parser.add_argument(
+        "-sg",
+        "--skip_good",
+        required=False,
+        action="store_true",
+        help="skip good models",
+    )
+    args = parser.parse_args()
+
+    if not os.path.exists(args.output_dir):
+        print(f"Error: Output directory '{args.output_dir}' does not exist.")
+        return sys.exit(1)
+
+    hf_model_repos = (
+        HF_LLM_MODLE_REPOS if not args.hf_model_repo else [(args.hf_model_repo,)]
+    )
+
+    device = "cpu"
+    prompt = "What is your favourite condiment?"
+    result_summary = {}
+
+    for repo_config in hf_model_repos:
+        hf_model_repo, status = repo_config[:2] + ("",) * (2 - len(repo_config))
+        if args.skip_good and status == "GOOD":
+            print(f"Skip model '{hf_model_repo}'.")
+            result_summary[hf_model_repo] = "SKIPPED"
+            continue
+
+        model_name = hf_model_repo.split("/")[-1].lower()
+        try:
+            print("\n\n#######################################################")
+            print(f"Load model '{hf_model_repo}' from Hugginface.")
+            print("#######################################################")
+            model = AutoModelForCausalLM.from_pretrained(
+                hf_model_repo, trust_remote_code=True
+            ).to(device)
+            model.eval()
+            tokenizer = AutoTokenizer.from_pretrained(
+                hf_model_repo, trust_remote_code=True
+            )
+            model_inputs = (
+                tokenizer([prompt], return_tensors="pt").to(device)["input_ids"],
+            )
+
+            edge_m = export_to_edge(
+                model,
+                model_inputs,
+                edge_compile_config=EdgeCompileConfig(
+                    _check_ir_validity=False,
+                ),
+            )
+            print(f"Exported '{hf_model_repo}' successfully.")
+
+            if args.to_et_w_xnn:
+                partitioners = {}
+                partitioners[
+                    XnnpackDynamicallyQuantizedPartitioner.__name__
+                ] = XnnpackDynamicallyQuantizedPartitioner()
+                prog = edge_m.to_backend(partitioners).to_executorch()
+                print(f"Lowered graph:\n{prog.exported_program().graph}")
+                filename = os.path.join(args.output_dir, f"xnnpack_{model_name}.pte")
+                with open(filename, "wb") as f:
+                    f.write(prog.buffer)
```

Comment:
@lucylq just landed a new API at https://github.com/pytorch/executorch/pull/2307/files#diff-006db5ac7da9220ba2c903d189243689ccfd41d23f27d363702dbc133c8187ecR483 that can significantly reduce the memory consumed during emitting/saving. To use it here, you could say
```
prog.write_to_file(f)
```
This will avoid allocating a single buffer that contains the entire .pte data; instead, it writes individual buffers to the file directly.